### PR TITLE
Add New Swift File command

### PR DESCRIPTION
**Pull request by plemarquand**
Add a command that creates a new Swift File. Add it as an option to the File > New File picker. Add it as an option when right clicking a folder in a Swift project context. Add a keybinding of Opt+S Opt+N (or Alt+S Alt+N) to create a new Swift file.

### CODE REVIEW

**Comment by plemarquand** [comment]
@swift-server-bot test this please

**Comment by plemarquand** [comment]
@swift-ci please test

File: src/commands/newFile.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    uri?: vscode.Uri,
+    isDirectory: (uri: vscode.Uri) => Promise<boolean> = async uri => {
+        return (await vscode.workspace.fs.stat(uri)).type === vscode.FileType.Directory;
+    }
+) {
+    if (uri) {
+        // Attempt to create the file at the given directory.
+        const dir = (await isDirectory(uri)) ? uri.fsPath : path.dirname(uri.fsPath);
+        const defaultName = path.join(dir, defaultFileName);
+        const givenPath = await vscode.window.showInputBox({
```
Comment (award999) [inline src/commands/newFile.ts:33]: https://code.visualstudio.com/api/references/vscode-api#window showOpenDialog or showSaveDialog is preferred way. User can enable simple file dialog in settings if they want to type out

File: test/suite/commands/NewFile.test.ts
Comment (award999) [inline test/suite/commands/NewFile.test.ts:1]: nit: NewFile inconsistent capitalization

File: test/suite/commands/NewFile.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VS Code Swift open source project
+//
+// Copyright (c) 2023 the VS Code Swift project authors
```
Comment (award999) [inline test/suite/commands/NewFile.test.ts:5]: 2024

File: src/commands/newFile.ts (hunk 1 above)
Comment (plemarquand) [reply src/commands/newFile.ts]: Much better, thanks

**Review by award999 — APPROVED** [review]

**Comment by award999** [comment]
Verified

**Pull request merged**